presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2103,12 +2108,13 @@ private DynamicFilterSourceOperatorFactory createDynamicFilterSourceOperatorFact
             }
             log.debug("[Join] Dynamic filters: %s", node.getDynamicFilters());
             LocalDynamicFiltersCollector collector = context.getDynamicFiltersCollector();
-            return LocalDynamicFilter
+            return LocalDynamicFilterConsumer
                     .create(node, partitionCount)
                     .map(filter -> {
                         // Intersect dynamic filters' predicates when they become ready,
                         // in order to support multiple join nodes in the same plan fragment.
-                        addSuccessCallback(filter.getResultFuture(), collector::intersect);
+                        context.registerDynamicFiltersConsumer(node.getId().toString(), filter);
```

Comment:
use `addSuccessCallback` instead of `registerDynamicFiltersConsumer`

Reply:
done